plugins/outputs/clarify/clarify.go

```diff
@@ -0,0 +1,186 @@
+//go:generate ../../../tools/readme_config_includer/generator
+
+package clarify
+
+import (
+	"context"
+	_ "embed"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/clarify/clarify-go"
+	"github.com/clarify/clarify-go/fields"
+	"github.com/clarify/clarify-go/views"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/outputs"
+)
+
+type Clarify struct {
+	Username        config.Secret   `toml:"username"`
+	Password        config.Secret   `toml:"password"`
+	CredentialsFile string          `toml:"credentials_file"`
+	Timeout         config.Duration `toml:"timeout"`
+	IDTags          []string        `toml:"id_tags"`
+	ClarifyIDTag    string          `toml:"clarify_id_tag"`
+	Log             telegraf.Logger `toml:"-"`
+
+	client *clarify.Client
+}
+
+var errIDTooLong = errors.New("id too long (>128)")
+
+const defaultTimeout = config.Duration(20 * time.Second)
+const allowedIDRunes = `abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890-_:.#+/`
+
+//go:embed sample.conf
+var sampleConfig string
+
+func (c *Clarify) Init() error {
+	if c.Timeout <= 0 {
+		c.Timeout = defaultTimeout
+	}
+	// Not blocking as it doesn't do any http requests, just sets up the necessarry Oauth2 client.
+	ctx := context.Background()
+	switch {
+	case c.CredentialsFile != "":
+		creds, err := clarify.CredentialsFromFile(c.CredentialsFile)
```

Comment:
I would really recommend to check if only the credential file is specified in order to error out early instead of surprises for users not reading the docs...
```suggestion
	case c.CredentialsFile != "":
		if !c.Username.Empty() || !c.Password.Empty() {
			return errors.New("only credential file OR username/password can be specified")
		}
		creds, err := clarify.CredentialsFromFile(c.CredentialsFile)
```

Reply:
Good idea. implemented.